Updating software-related terminologies

jagru20 opened the pull request:
- Added schemas for a few more terms.
- Added Links to ontology for existing Terms where possible.

jagru20 committed:
Update Terminologies

Added Links to ontology for the software-related controlledTerms

jagru20 committed:
Update Terminologies

Added newline at the end of each file

jagru20 committed:
Update dataProcessing.jsonld

Fix link

jagru20 committed:
Update and rename 3DGeometryData.jsonld to 3DGeometryDataTypes.jsonld

canged name and added ontology Identifier

jagru20 committed:
Create support

jagru20 committed:
Rename support to support.jsonld

jagru20 committed:
Update and rename support.jsonld to softwareSupport.jsonld

changed the contributiontype name into softwareSupport to allow clear differentiation from other forms of support.

lzehl commented on terminologies/contributionType/softwareSupport.jsonld:
```diff
@@ -1,7 +1,7 @@
-  "name": "support",
+  "name": "softwareSupport",
   "definition": null,
   "description": "Contributing by providing support to users of the product.",
```
If you rename it to softwareSupport the description needs to be specific as well. And we need to introduce other specific support contribution as well (e.g., technicalSupport, scientificSupport?, etc).

lzehl reviewed (changes requested):
@jagru20 thanks a lot for updating / extending the software parts in the controlledTerms. It's highly appreciated :slightly_smiling_face: 

A general feedback point: please check again all "names" in your changes. The entries here should be with space and everything. The lowerCamelCase convention needs to be only followed for the properties, and the UpperCamelCase convention for the types. But the values can / should be written normally.

Please, see also my comment for the support.json-ld

Commit:
changed names to normal case

lzehl reviewed (changes requested):
@jagru20 I think there was a misunderstanding between us for the "name" entries. They do not need to use lowerCamelCase or UpperCamelCase, but they should be written in lower capitals except for abbreviations (e.g. HPC; although abbreviations should only be used in "name" if they are more commonly known than the full name) and given names (e.g., Java). Could you update this accordingly?

Otherwise this PR looks good.

jagru20 commented:
> 
> 
> @jagru20 I think there was a misunderstanding between us for the "name" entries. They do not need to use lowerCamelCase or UpperCamelCase, but they should be written in lower capitals except for abbreviations (e.g. HPC; although abbreviations should only be used in "name" if they are more commonly known than the full name) and given names (e.g., Java). Could you update this accordingly?
> 
> Otherwise this PR looks good.

Yes, I got that wrong then. I will update that.

lzehl commented:
@jagru20 top :) thanks a lot!

jagrue committed:
cleanup:
deleted softwareSupport.jsonld as it is covered by supportChannel in researchProduct Version now
changed names to lowercase
added some missing ontology identifiers
